packages/services/src/terminal/default.ts

```diff
@@ -228,9 +237,35 @@ class DefaultTerminalSession implements TerminalSession.ISession {
           reject(event);
         }
       };
+
+      socket.onclose = (event: CloseEvent) => {
+        console.error(`Terminal websocket closed: ${event.code}`);
+        this._reconnectSocket();
+      };
     });
   }
 
+  private _reconnectSocket = (): void => {
+    if (this._isDisposed || !this._ws) {
+      return;
+    }
+    if (this._autoReconnectAttempt < this._autoReconnectLimit) {
+      this._isReady = false;
+      let timeout = Math.pow(2, this._autoReconnectAttempt);
+      console.error('Terminal websocket reconnecting in ' + timeout + ' seconds.');
+      setTimeout(() => {
+        this._initializeSocket().then(() => {
+          console.error('Terminal websocket reconnected');
+        }).catch((e) => {
+          console.error(`Terminal websocket reconnecting error`);
+        });
+      }, 1e3 * timeout);
```

Comment:
I would suggest moving the `1e3 * timeout` up to where timeout is declared for clarity.